cognite/neat/core/extractors/graph_sheet_to_graph.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import warnings
+
+import pandas as pd
+from rdflib import RDF, Graph, Literal
+
+from cognite.neat.core.data_classes.transformation_rules import TransformationRules
+
+
+def validate_if_graph_capturing_sheet_empty(graph_capturing_sheet: dict[str, pd.DataFrame]):
+    """Validate if the graph capturing sheet is empty
+
+    Parameters
+    ----------
+    graph_capturing_sheet : dict[str, pd.DataFrame]
+        Graph capturing sheet
+    """
+    if all(df.empty for df in graph_capturing_sheet.values()):
+        logging.error("Graph capturing sheet is empty! Aborting!")
+        raise ValueError("Graph capturing sheet is empty! Aborting!")
+
+
+def validate_rules_graph_pair(graph_capturing_sheet: dict[str, pd.DataFrame], transformation_rule: TransformationRules):
+    """Validate if the graph capturing sheet is based on the transformation rules
+
+    Parameters
+    ----------
+    graph_capturing_sheet : dict[str, pd.DataFrame]
+        Graph capturing sheet
+    transformation_rule : TransformationRules
+        Transformation rules
+    """
+    intersection = set(graph_capturing_sheet.keys()).intersection(set(transformation_rule.get_defined_classes()))
+
+    if not intersection:
+        logging.error("Graph capturing sheet is not based on transformation rules! Aborting!")
+        raise ValueError("Graph capturing sheet is not based on transformation rules! Aborting!")
+
+    elif len(intersection) == len(graph_capturing_sheet.keys()):
+        logging.info("All classes in the graph capturing sheet are defined in the transformation rules!")
+
+    elif len(intersection) < len(graph_capturing_sheet.keys()):
+        logging.warning(
+            "Graph capturing sheet contains classes that are not defined in the transformation rules! Proceeding..."
+        )
+        warnings.warn(
+            "Graph capturing sheet contains classes that are not defined in the transformation rules! Proceeding...",
+            stacklevel=2,
+        )
+
+    elif len(intersection) < len(transformation_rule.get_defined_classes()):
+        logging.warning(
+            "Transformation rules contain classes that are not present in the graph capturing sheet! Proceeding..."
+        )
+        warnings.warn(
+            "Transformation rules contain classes that are not present in the graph capturing sheet! Proceeding...",
+            stacklevel=2,
+        )
+
+
+def sheet2graph(
+    graph_capturing_sheet: dict[str, pd.DataFrame], transformation_rule: TransformationRules, separator: str = ","
+):
+    """Converts a graph capturing sheet to an RDF graph
+
+    Parameters
+    ----------
+    graph_capturing_sheet : dict[str, pd.DataFrame]
+        Graph capturing sheet
+    transformation_rule : TransformationRules
+        Transformation rules
+    separator : str, optional
```

Comment:
```suggestion
    separator: str, optional
```

Reply:
This is default way for numpydocstring:
https://numpydoc.readthedocs.io/en/latest/format.html#parameters